workflows/sra_download.nf

```diff
@@ -4,6 +4,16 @@
 ========================================================================================
 */
 
+def valid_params = [
+    ena_metadata_fields : ['run_accession', 'experiment_accession', 'library_layout', 'fastq_ftp', 'fastq_md5']
```

Comment:
I got lost here, this means that only these 5 fields can be retrieved?

Reply:
Nope. The code [here](https://github.com/drpatelh/rnaseq/blob/eb32d11b1cbcf3bc25fe355bcf01dbc92f237899/lib/WorkflowSraDownload.groovy#L13) just checks that `--ena_metadata_fields` contains those 5 fields as a minimum but it can contain as many others as long as they exist [here](https://www.ebi.ac.uk/ena/portal/api/returnFields?dataPortal=ena&format=tsv&result=read_run) 

Was tricky figuring out how best to do this because if `--ena_metadata_fields` is not specified then by default the fields [here](https://github.com/drpatelh/rnaseq/blob/eb32d11b1cbcf3bc25fe355bcf01dbc92f237899/bin/sra_ids_to_runinfo.py#L18-L30) will be used but I didn't want to specify a long list list that in the `nextflow_schema.json`  so this was the workaround I came up with. I also manually curated the default list to remove any blank metadata fields so that accessions etc are next to each other in the final samplesheet so makes it easier to organise and visualise.

Reply:
Makes a lot of sense! Thanks for the explanation 😄